### ISSUE TITLE
dict.get

### ISSUE THREAD

**Issue by jayvdb** [adsharma/py2many#293]
Python:
```py
def dict_get():
    CODES = {"KEY": 1}
    assert CODES.get("KEY") == 1


def dict_get_default():
    CODES = {"KEY": 1}
    assert CODES.get("MISSING", 2) == 2
```
Kotlin works with the former only, and Julia with the latter only.  All others fail.

Also we're going to run into nullable problems with `.get(x)` which returns None, e.g. in Dart which uses `a[x]` and returns nullable `String?`. In Dart we can avoid the nullable if there is a default with `map[x] ?? default`, but if default is not a constant then we need to track the variable to determine if is nullable or not.  Likely needs a nullable annotation like `lifetime`, but please lets try to disable "sound null" in dart and equivalent in other languages and not tackle nullability until there is a good sample of viable test cases to ensure it is sufficiently implemented that it isnt preventing progress.

Transpiles into Rust with errors:
```rust
error[E0605]: non-primitive cast: `Option<&i32>` as `i32`
  --> dict.rs:45:13
   |
45 |     assert!((CODES.get("KEY") as i32) == 1);
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^ an `as` expression can only be used to convert between primitive types or to coerce to a specific trait object

error[E0061]: this function takes 1 argument but 2 arguments were supplied
   --> dict.rs:50:20
    |
50  |     assert!((CODES.get("MISSING", 2) as i32) == 2);
    |                    ^^^ ---------  - supplied 2 arguments
    |                    |
    |                    expected 1 argument
    |
note: associated function defined here

error[E0605]: non-primitive cast: `Option<&i32>` as `i32`
  --> dict.rs:50:13
   |
50 |     assert!((CODES.get("MISSING", 2) as i32) == 2);
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ an `as` expression can only be used to convert between primitive types or to coerce to a specific trait object
```

**Comment by adsharma** [comment]
Yeah - the main thing that's missing here is that transpiler doesn't understand the semantics of python's `dict.get()`.
I feel that at ~1000 lines of code $lang/transpiler.py is at around the limits of maintainability. Adding more code in there to make it understand `dict.get()` semantics (and other common python idioms) is a significant amount of complexity.

I like adding plugins.py to all languages (currently only rust has it) and handling it there. The key idea is to index any tables there with `dict.get` instead of strings.
